content/advent-2016/sea-of-open-source.md

```diff
@@ -0,0 +1,165 @@
++++
+author = ["Florin Pățan"]
+date = "2016-12-15T00:00:00+00:00"
+title = "Finding good packages in the Sea of Open-Source"
+series = ["Advent 2016"]
++++
+
+So far we've seen different ways to use Go to build a variety of applications,
+from [versioning your data pipelines](https://blog.gopheracademy.com/advent-2016/pachyderm/) to building your own [BBQ grill controller](https://blog.gopheracademy.com/advent-2016/qpid/).
+
+But often times you'll be looking to reuse the code from others in order to
+allow you to focus on what your code should do and not reinvent the wheel.
+
+How do you know which package to use? And when should you use a package
+versus writing your own?
+
+Let that sink in for a bit and now think of when you are learning a language
+and you don't really know what is considered idiomatic or not.
+
+Those are hard questions. They become even harder if you have to use these
+third-party packages to build applications that are going to be used into
+production code.
+
+Usually you have a checklist of items that you'd like a package or library
+to meet before you can use it. My list looks like this (not in the order of 
+importance):
+
+- is it used by others?
+- how does the documentation look like?
+- does it do what I need it to do (and preferably nothing more)?
+- is the author maintaining the package? How are issues / PRs addressed? Are
+there any contributors to the project?
+- license
+- what's the code quality like? Does it have tests?
```

Comment:
and do the tests pass? :-)

This is huge for me, because having to fix failing/flaky tests before contributing a change to a project is a deterrent.